### DISCUSSION TITLE
Filter for sql queries

### DISCUSSION THREAD

**Discussion by MikhnoRuslan** [abpframework/abp#20016] [Ideas]
Hi everyone.
It will be nice to implement some generic method for filtering queries (and list for sure). Currently we can use WhenIf(). But we can add string fild into input and then parse value and make a filter.

for example: id~eq~1~and~name~eq~test~or~test2~and~userDetails.status~gt~2 and so on. After parse we can get expression Where(like x => x.Id == 1 && x.Name == test || x.Name == test2 && x.UserDetail.Status > 2)

**Comment by maliming** [comment]
hi

This doesn't seem related to the abp framework.
